src/ringbuffer.rs

```diff
@@ -70,16 +71,15 @@ impl<T: Slot + Send> RingBuffer<T> {
     // Unsafe because we have no guarantees the caller won't invalidate this slot
     pub unsafe fn get(&self, from: usize, size: usize) -> &[T] {
         debug!("              RingBuffer get({}, {})", from, size);
-        let v : *mut Vec<T> = self.entries.get();
-        ::std::slice::from_raw_parts(v.offset(from), size)
+        &self.entries.get().as_ref().unwrap()[from .. from + size]
     }
 
     // Unsafe because we have no guarantees the caller won't invalidate this slot
     pub unsafe fn write(&self, position: usize, data: T) {
-        let v: *mut Vec<T> = self.entries.get();
-        let slot = v.get_mut(position);
-        drop(slot);
-        slot = data;
+        let v = self.entries.get();
+        let mut slot = v.as_mut().unwrap().get_mut(position).unwrap();
+        //TODO: Do we need to `drop` the current data?
```

Comment:
So the "correct" version would be unsafe { self.entries.get().as_mut().unwrap()[pos] = data; }

But that would require `#![feature(ptr_as_ref)]` and run only on nightly. An alternative would be to `push(data); swap_remove(position)`, but that would make a second copy, which is likely to negatively impact performance.